test-integration/DepositZap.js

```diff
@@ -0,0 +1,347 @@
+      addressRegistry.address,
+      tvlAgg.address,
+      [CURVE_3CRV_ADDRESS],
+      [USDC_AGG_ADDRESS],
+      86400,
+      86400
+    );
+    await oracleAdapter.deployed();
+    await addressRegistry.registerAddress(
+      bytes32("oracleAdapter"),
+      oracleAdapter.address
+    );
+  });
+
+  before("Deploy Index Token", async () => {
+    const IndexToken = await ethers.getContractFactory("TestIndexToken");
+    const logic = await IndexToken.deploy();
+    await logic.deployed();
+
+    const initData = IndexToken.interface.encodeFunctionData(
+      "initialize(address,address)",
+      [addressRegistry.address, curve3Crv.address]
+    );
+    const proxy = await deployProxy(logic, proxyAdmin, initData);
+    indexToken = await IndexToken.attach(proxy.address);
+  });
+
+  before("Deploy Deposit Zap", async () => {
+    const DepositZap = await ethers.getContractFactory("DepositZap");
+    depositZap = await DepositZap.deploy(indexToken.address);
+    await depositZap.deployed();
+  });
+
+  describe("deposit", () => {
+    it("can deposit DAI", async () => {
+      expect(await indexToken.balanceOf(randomUser.address)).to.equal(0);
+
+      await acquireToken(
+        WHALE_POOLS["DAI"],
+        randomUser.address,
+        dai,
+        "1000000",
+        randomUser.address
+      );
+
+      const depositAmount = tokenAmountToBigNumber(1000, 18);
+      const index = 0; // DAI
+
+      await dai.connect(randomUser).approve(depositZap.address, depositAmount);
+      await depositZap.connect(randomUser).deposit(depositAmount, index);
+
+      expect(await indexToken.balanceOf(randomUser.address)).to.gt(0);
+    });
+
+    it("can deposit USDC", async () => {
+      expect(await indexToken.balanceOf(randomUser.address)).to.equal(0);
+
+      await acquireToken(
+        WHALE_POOLS["USDC"],
+        randomUser.address,
+        usdc,
+        "1000000",
+        randomUser.address
+      );
+
+      const depositAmount = tokenAmountToBigNumber(1000, 6);
+      const index = 1; // USDC
+
+      await usdc.connect(randomUser).approve(depositZap.address, depositAmount);
+      await depositZap.connect(randomUser).deposit(depositAmount, index);
+
+      expect(await indexToken.balanceOf(randomUser.address)).to.gt(0);
+    });
+
+    it("can deposit Tether", async () => {
+      expect(await indexToken.balanceOf(randomUser.address)).to.equal(0);
+
+      await acquireToken(
+        WHALE_POOLS["USDT"],
+        randomUser.address,
+        tether,
+        "1000000",
+        randomUser.address
+      );
+
+      const depositAmount = tokenAmountToBigNumber(1000, 6);
+      const index = 2; // Tether
+
+      await tether
+        .connect(randomUser)
+        .approve(depositZap.address, depositAmount);
+      await depositZap.connect(randomUser).deposit(depositAmount, index);
+
+      expect(await indexToken.balanceOf(randomUser.address)).to.gt(0);
+    });
+  });
+
+  describe("redeem", () => {
```

Comment:
While it may not be relevant in practice, would be good to have a test for a scenario where there is no exit liquidity in the 3pool and a user is trying to redeem. Similarly, if a user is trying to redeem and there is no 3crv in the `IndexToken` reserve.